### PR TITLE
Allocations: New Budget panel

### PR DESCRIPTION
**Pull request by PeterMPhillips**
~When the `budget-detail` branch (PR #1758) gets merged I'll redirect this to `dev`~

Small visual changes to the panel:
Screenshot from 2019-12-12 20-44-17

And here it shows weeks if the number of days is a multiple of 7:
Screenshot from 2019-12-12 20-44-57

### CODE REVIEW

File: apps/allocations/app/components/Panel/NewBudget.js [hunk 1]
```diff
@@ -139,14 +146,33 @@ class NewBudget extends React.Component {
...
+  const days = Math.round(ms/86400000)
+  if(days%7 === 0){
+    const weeks = days/7
+    return `${weeks === 1 ? 'week' : `${weeks} weeks`}`
+  }
+  return `${days === 1 ? 'day' : `${days} days`}`
+}
+
 const NewBudgetWrap = props => {
   const { appState: { tokens = [] } } = useAragonApi()
```
Comment (chadoh) [inline apps/allocations/app/components/Panel/NewBudget.js]: Let's move the `duration` calculation here, please!

File: apps/allocations/app/components/App/Overview.js [hunk 2]
```diff
@@ -11,14 +11,21 @@ import useSaveBudget from '../../hooks/useSaveBudget'
 
 const Overview = () => {
   const { appState } = useAragonApi()
-  const { allocations = [], budgets = [], isSyncing = true } = appState
+  const { allocations = [], budgets = [], period, isSyncing = true } = appState
   const { setPanel } = usePanel()
   const saveBudget = useSaveBudget()
 
+  //Set duration to default, then calculate if 'period' is set
+  let duration = 2592000000
```
Comment (chadoh) [inline apps/allocations/app/components/App/Overview.js]: We can move this outside of the component. And this number is incomprehensible to me; I'd love to see it shown as a calculation. 

```js
const DEFAULT_DURATION = 30 * 24 * 60 * 60 * 1000

File: apps/allocations/app/components/Panel/NewBudget.js [hunk 3]
```diff
@@ -139,14 +146,33 @@ class NewBudget extends React.Component {
                 onChange={this.handleSelectToken}
               />
             )}
-
+            <Text
+              css={`
+                white-space: nowrap;
+                line-height: 40px;
+                padding-left: 16px;
```
Comment (chadoh) [inline apps/allocations/app/components/Panel/NewBudget.js]: We should probably use `padding-left: ${2 * GU}px` here

File: apps/allocations/app/components/Panel/NewBudget.js [hunk 4]
```diff
@@ -139,14 +146,33 @@ class NewBudget extends React.Component {
...
+            </Text>
           </InputGroup>
         </Field>
       </Form>
     )
   }
 }
 
+const timeString = ms => {
+  const days = Math.round(ms/86400000)
```
Comment (chadoh) [inline apps/allocations/app/components/Panel/NewBudget.js]: I'd prefer we expand this large number into its constituent pieces, too – we could set up a constant, if you want. `ms / MS_PER_DAY`, where `const MS_PER_DAY = 24 * 3600 * 1000` is set at the top of the file

File: apps/allocations/app/components/Panel/NewBudget.js [hunk 5]
```diff
@@ -139,14 +146,33 @@ class NewBudget extends React.Component {
...
+  if(days%7 === 0){
+    const weeks = days/7
+    return `${weeks === 1 ? 'week' : `${weeks} weeks`}`
+  }
+  return `${days === 1 ? 'day' : `${days} days`}`
```
Comment (chadoh) [inline apps/allocations/app/components/Panel/NewBudget.js]:
```js
return days === 1 ? 'day' : `${days} days`
```

Seems cleaner to get rid of the wrapping backticks/interpolation

**Comment by PeterMPhillips** [comment]
@chadoh I turned the timeString() function into a component called Duration. It calls the useAragonApi and useTheme hooks so they don't have to passed as props. Let me know if that works for you, happy to amend it if its not to your liking

File: apps/allocations/app/components/Panel/NewBudget.js [hunk 6]
```diff
@@ -139,29 +148,53 @@ class NewBudget extends React.Component {
...
+      `}
+    >
+      every {string}
+    </Text>
   )
 }
 
+const NewBudgetWrap = props => {
+  const { appState: { tokens = [] } } = useAragonApi()
+  return <NewBudget tokens={tokens} {...props} />
```
Comment (chadoh) [inline apps/allocations/app/components/Panel/NewBudget.js]: Why stop passing in `saveBudget` and `theme`? This seems like a bug.

Comment (PeterMPhillips) [reply apps/allocations/app/components/Panel/NewBudget.js]: I think that `saveBudget` must have been removed when I rebased. But I believe it gets passed in `{...props}', no?

`theme` was only being used by a `Text` component that has been moved to `Duration` so `NewBudget` doesn't need it anymore

Comment (chadoh) [reply apps/allocations/app/components/Panel/NewBudget.js]: `saveBudget` was being defined in `NewBudgetWrap` and passed into `NewBudget`. I think you'll find that saving budgets no longer works.

**Pull request merged**